pkg/controller/datavolume-controller.go

```diff
@@ -916,6 +975,40 @@ func cloneStrategyToCloneType(selectedCloneStrategy cloneStrategy) string {
 	return ""
 }
 
+// reconcileSizelessClone handles the reconciling process if the size-detection mechanism is not able to obtain the img size
+func (r *DatavolumeReconciler) reconcileSizelessClone(
```

Comment:
When seeing this function this was screaming, use me in all 3 variations.
The csi clone, the snapshot clone, and the no target size clone. I would call it something like `isSourceReadyToClone` that returns a boolean and error. The error is to catch the normal errors like now. When calling this function I would do it something like this for the csi clone and smart clone options.
```go
if readyToClone, err := r.isSourceReadyToClone(dv, cloneStrategy); err != nil {
  return reconcile.Result{}, err
} else if !readyToClone {
  return reconcile.Result{Requeue: true},
    r.updateCloneStatusPhase(cdiv1.CloneScheduled, datavolume, nil, selectedCloneStrategy)
}
```
For the size less clone option something like this
```go
else if !done {
  if readyToClone, err := r.isSourceReadyToClone(dv, cloneStrategy); err != nil {
    return reconcile.Result{}, err
  } else if !readyToClone {
    return reconcile.Result{Requeue: true},
      r.updateCloneStatusPhase(cdiv1.CloneScheduled, datavolume, nil, cloneStrategy)
  }
  return reconcile.Result{}, nil
}
```